firehose/parsers/flawfinder.py

```diff
@@ -0,0 +1,149 @@
+#
+#   You should have received a copy of the GNU Lesser General Public
+#   License along with this library; if not, write to the Free Software
+#   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301
+#   USA
+
+import sys
+import re
+from subprocess import check_output
+from firehose.model import Message, Point, \
+    File, Location, Generator, Metadata, Analysis, Issue
+
+
+def main():
+    """ Main entry to flawfinder parser """
+    arg_file = sys.argv[1]
+    report = load_file(arg_file)
+    analysis = parse_file(report)
+    sys.stdout.write(str(analysis.to_xml()))
+    sys.stdout.write('\n')
+
+def load_file(arg_file):
+    """ Loads the Flawfinder report in memory
+
+    :arg1: file passed by cli
+    :returns: file loaded in memory
+    """
+    try:
+        return open(arg_file, 'r')
+    except Exception as e:
+        raise e
+
+
+def parse_file(loaded_file):
+    """ Parser loaded flawfinder output
+
+    :loaded_file: flawfinder report, loaded in memory
+    :returns: Firehose Analysis object, representing the final XML.
+    """
+
+    generator = Generator(name='flawfinder',
+                          version=flawfinder())
+    metadata = Metadata(generator, None, None, None)
+    analysis = Analysis(metadata, [])
+
+    line = loaded_file.readline()
+
+    location_nodes = list()
+    weakness_paths = []
+    weakness_lines = []
+    weakness_messages = []
+    weakness_cwes = []
+    pattern = "\([a-z]*\)"
+    pattern2 = "\[[0-9]\]"
+    prog = re.compile(pattern)
+    prog2 = re.compile(pattern2)
+
+    while line:
+        if line != "\n" and (prog.search(line) and prog2.search(line)):
+            cwe = ""
+            weakness_paths.append(line.split(":")[0])
+            weakness_lines.append(line.split(":")[1])
+
+            message_line = loaded_file.readline()
+            message = ""
+            while message_line != "\n" and not \
+            (prog.search(message_line) and prog2.search(message_line)):
+                message += " " + message_line
+                extracted_cwe = get_cwe(message_line)
+                if extracted_cwe != '':
+                    cwe = extracted_cwe
+                message_line = loaded_file.readline()
+            weakness_messages.append(message)
+            weakness_cwes.append(cwe)
+            line = message_line
+        else:
+            line = loaded_file.readline()
+
+    counter = 0
+    for weakness in weakness_paths:
+        location = Location(file=File(weakness, None),
+                            function=None,
+                            point=Point(int(weakness_lines[counter]), 0))
+
+        cwe = weakness_cwes[counter]
+        if cwe == '':
+            cwe = None
+        else:
+            cwe = int(weakness_cwes[counter])
+
+        issue = Issue(cwe, None, location,
+                      Message(text=weakness_messages[counter]), notes=None,
+                      trace=None, severity=None, customfields=None)
```

Comment:
Re `severity`: Flawfinder's docs say:
> The risk level is shown inside square brackets and varies from 0, very little risk, to 5, great risk.

It would be good to capture that number as `Issue.severity` (as a string, without the square brackets):
http://firehose.readthedocs.io/en/latest/data-model.html#firehose.model.severity